minerl/Malmo/Minecraft/src/main/java/com/microsoft/Malmo/MissionHandlers/ObservationFromFullInventoryImplementation.java

```diff
@@ -175,44 +174,32 @@ public static String getInventoryName(IInventory inv)
 
     public static void getInventoryJSON(JsonArray arr, IInventory inventory)
     {
-        String invName = getInventoryName(inventory);
         for (int i = 0; i < inventory.getSizeInventory(); i++)
         {
             ItemStack is = inventory.getStackInSlot(i);
-            if (is != null)
-            {
-                JsonObject jobj = new JsonObject();
-                DrawItem di = MinecraftTypeHelper.getDrawItemFromItemStack(is);
-                String name = di.getType();
-                if (di.getColour() != null)
-                    jobj.addProperty("colour", di.getColour().value());
-                if (di.getVariant() != null)
-                    jobj.addProperty("variant", di.getVariant().getValue());
-                jobj.addProperty("type", name);
-                jobj.addProperty("index", i);
-                jobj.addProperty("quantity", is.getCount());
-                jobj.addProperty("inventory", invName);
-                arr.add(jobj);
-            }
+            JsonObject jobj = MineRLTypeHelper.jsonFromItemStack(is);
+            jobj.addProperty("index", i);
+            jobj.addProperty("inventory", getInventoryName(inventory));
+            arr.add(jobj);
         }
     }
 
-    public static void getInventoryJSON(JsonObject json, String prefix, IInventory inventory, int maxSlot)
+    public static void getInventoryJSON(JsonObject json, String prefix, @NotNull IInventory inventory, int maxSlot)
     {
         int nSlots = Math.min(inventory.getSizeInventory(), maxSlot);
         for (int i = 0; i < nSlots; i++)
         {
             ItemStack is = inventory.getStackInSlot(i);
+            // DEBUG conditional
+            if (is == null) {
+                throw new RuntimeException(
+                        String.format("Unexpected null in inventory at index %d.", i));
+            }
+
             if (is != null)
             {
                 json.addProperty(prefix + i + "_size", is.getCount());
-                DrawItem di = MinecraftTypeHelper.getDrawItemFromItemStack(is);
-                String name = di.getType();
-                if (di.getColour() != null)
-                    json.addProperty(prefix + i + "_colour",  di.getColour().value());
-                if (di.getVariant() != null)
-                    json.addProperty(prefix + i + "_variant", di.getVariant().getValue());
-                json.addProperty(prefix + i + "_item", name);
+                json.addProperty(prefix + i + "_item", MineRLTypeHelper.getItemType(is.getItem()));
```

Comment:
Metadata could be added here

Reply:
We never actually use this part of the code -- the line that calls this function claims that this is for "backwards compatibility".

Reply:
"metadata" field is written in by the non-legacy function that calls `MineRLTypeHelper.jsonFromItemStack(is)` above.